lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -719,17 +718,23 @@ RequirementMatch swift::matchWitness(TypeChecker &tc,
     }
   }
 
-  Optional<RequirementEnvironment> reqEnvironment(
-      RequirementEnvironment(dc, reqSig, proto, covariantSelf, conformance));
+  WitnessChecker::RequirementEnvironmentCacheKey cacheKey(reqSig,covariantSelf);
```

Comment:
Maybe extract this out into a getOrCreateRequirementEnvironment(). Also I don't like this trend of omitting spaces to get things to fit in 80 characters. Either live with 81 characters or break the line :)